shared/rest/src/main/java/io/pravega/shared/rest/RESTServer.java

```diff
@@ -73,7 +73,7 @@ protected void startUp() {
                 contextConfigurator.setKeyStorePass(JKSHelper.loadPasswordFrom(restServerConfig.getKeyFilePasswordPath()));
                 httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true,
                         new SSLEngineConfigurator(contextConfigurator, false, false, false)
-                                .setEnabledProtocols(TLSProtocolVersion.getTlsProtocolVersionList(restServerConfig.tlsProtocolVersion())));
+                                .setEnabledProtocols((new TLSProtocolVersion(restServerConfig.tlsProtocolVersion())).tlsProtocols().toArray(new String[0])));
```

Comment:
There is a series of problems still.  

1. the fact tlsProtocols() returns a list has now made  the line above veyr strange with additional conversations etc.  Why can't  tlsProcotols() simply return a String[] which can be natively consumed here ?
2. We have 3 copies of TlsProtocolsVersion class.   Can it not be in one shared placed, used by all 3 components that need it?
3. If you look at this line and the others in Segmentstore and grpc,  you are still taking the raw String of tlsprotocols and parsing it here (even though it's buried inside the constructor/function calls).   I am wondering why TLSProtocolsVersion cannot be instantiated (and therefore parsed) once (once per config type; grpc, rest, and segment store), and passed around as is.   Meaning in this function let's say, an object of class TLSProtocolVersion is passed into here, rather than instantiated here based on restServerConfig.tlsProtocolVersion.   Have restServerConfig.tlsProtocol BE of type TLSProtocolsVersion.   Is that possible ?  (same comment in the other 2 configs)

Reply:
1. It was modified to `List` to make it immutable because of `spotbug` error, but removed this entire List and now a simple String[] is being returned.
2. Removed duplication and moved it to a shared place in `io/pravega/common/security` which is being used by all 3 components.
3. Used only ENUM as Java supports it without a Class so this avoid instantiation.